# **신용카드 사용자 연체 프로젝트**
# **Stage 1**

# **1. Intro**

**'내 집 마련'**　누구나 마음 한켠에서 고민하는 주제입니다.<br/>
- *통계청 2015년 자료에 의하면 (https://bit.ly/2SFyzMA)*    
*일반적인 한국인의 절반은 48.1%는 아파트에 살고 있습니다.*     
*또한 아파트를 부의 증식 수단으로 생각 하기 때문에 누구나 아파트 가격에 관심이 많습니다.*  
*하지만 자고 일어나면 달라지는 아파트 가격, 언제 집을 구매하면 좋을까요? *  
*초보부터 고수까지 모두가 참여하는 아파트 실거래가 예측 프로젝트를 소개합니다!*    



<p style='text-align: right;'> January, 2023, ver.001 | DaCrew 4기 TEAM 먼스크루</p>

---

#### <font color="midnightblue">**아파트 실거래가 예측 프로젝트**</font>에 참여하신 데이커 여러분, 환영합니다. 🙌

신나는 퇴근길, 서울 야경을 보면 저렇게 많은 불빛 가운데 내 집은 없다는 사실에 고민이 많아집니다.🥲<br/>
하지만 하루가 다르게 자고 일어나면 달라지는 부동산 가격, <br/>
높은 가격에 구매하기 싫은 구매자의 심리는 우리를 고민속에 빠지게 만듭니다.<br/>
이번 교육을 통해 **아파트 매매 데이터**로 아파트 매매 가격을 예측해 내는 멋진 데이커🥇가 되어 보세요!

---




# **2. 프로젝트 배경**

최근 뉴스를 보면 부동산 가격이 떨어지고 있음을 알 수 있습니다. <br>
하지만 우리의 기억 속에는 1년 전 치솟는 가격속에 부동산 열풍이 기억납니다. <br>
생각해보면 우리는 가격이 오르면 떨어지기를 기다리고 정작 가격이 떨어지면 최저점이 될 때 까지 기다리다 시기를 놓치곤 합니다. <br>
매번 남이 사면 오르고 내가 사면 떨어지는 마법같은 가격의 비밀을 데이터를 통해 합리적인 가격을 예측해 보고자 해당 프로젝트를 만들었습니다.


# **3. 데이터 살펴보기**

## 3.1. 데이터 불러오기

데이터 분석을 위한 첫 관문은 **데이터를 사용할 수 있도록 불러오는 것입니다.**    
불러온 데이터를 살펴보며 데이터의 형식과 대략적인 특징을 파악하고, 어떤 분석이 가능한지 확인하여 방향을 잡는 과정이 필요합니다.


---



이번 프로젝트의 데이터는 총 5가지이며 모두 csv 파일입니다.

> 1. **train.csv** : 서울/부산 지역의 1,100,000여개 거래 데이터, 아파트 거래일, 지역, 전용면적, 실 거래가 등의 정보
> 2. **test.csv** : 실 거래가를 제외하고 train.csv와 동일
> 3. **park.csv** : 서울/부산 지역의 공원에 대한 정보
> 4. **day_care_center.csv** : 서울/부산 지역의 어린이집에 대한 정보 정답 파일
> 5. **submission.csv** : 제출 대상 파일

분석에 사용할 데이터 파일을 불러오기 위해, **pandas 패키지**가 가지고 있는 **<font color="orange">read_csv()</font>** 함수를 사용하겠습니다.   
- pandas 패키지에 관한 내용은 다음 링크를 참고해주세요! [pandas 공식 유저 가이드](https://pandas.pydata.org/docs/user_guide/index.html#user-guide)    
- 패키지 불러오기: **<font color="orange">import 패키지명 as 약어</font>**

---







### Inst.

1. pandas 패키지를 pd 라는 약칭으로 import 하세요.

2. read_csv() 메소드를 이용해    
<font color="green">train.csv, test.csv, park.csv, day_care_center.csv, sample_submission.csv </font>파일을    
<font color="green">train, test, park, day_care_center, sample_submission</font> 이름으로 불러오세요!

In [ ]:
# 1. pandas 패키지를 pd 라는 약칭으로 import 하세요.


In [ ]:
# 2. read_csv() 메소드를 이용해 파일 불러오세요.


### Hint.
- <font color="orange">약칭.메소드</font>로 패키지를 사용할 수 있습니다.
    - <font color="orange">OO.OOOO_OOO()</font>

- 파일을 불러올 변수 = 약칭.read_csv("불러올 파일 경로")


### Solution.
**데이터를 불러오는 방식은 접속 환경(로컬 컴퓨터, Colab 등)에 따라 다양**하나, [Jupyter Lite](https://jupyter.org/try-jupyter/lab/)에서 불러올 수 있는 방법을 소개하겠습니다.    
<img src='http://drive.google.com/uc?export=view&id=165LPD276TSlFXZYjFSL5JKlIlYiZwszc' /><br>
왼쪽 위에서 파일 업로드 아이콘을 선택하여 데이터 5개를 업로드합니다.

In [1]:
# 1. pandas 패키지를 pd 라는 약칭으로 import 하세요.
import pandas as pd # import 패키지명 as 약어

In [4]:
from google.colab import drive
import pandas as pd

drive.mount('/content/drive')

PATH = "/content/drive/MyDrive/대외활동/데이크루6기_보초석분/data/"

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [8]:
# 2. read_csv() 메소드를 이용해 파일 불러오세요.
train = pd.read_csv(PATH+'train.csv') # pd.read_csv(PATH) -> PATH에 로컬 저장 경로 입력 ex) C:/home/data/train.csv
test = pd.read_csv(PATH+'test.csv')
submission = pd.read_csv(PATH+'sample_submission.csv')

## 3.2. 데이터 확인

데이터가 잘 불러와졌는지, 어떻게 생겼는지 살펴보겠습니다.

데이터 과학 실무 과정에서는 데이터를 살펴볼 때, **이상치나 결측치**가 존재하는지 알기 위하여    
**head()**, **info()** 메소드를 활용하는 경우가 많습니다.

그럼 데이터가 어떻게 생겼는지 이 두 가지 메소드를 이용하여 직접 확인해 보세요!

### Inst.
1.   **head()** 함수를 사용하여 train, test, submission 데이터의 **처음 5행**을 확인해 보세요.
2.   **info()** 함수를 사용하여 train, test, submission 데이터의  **결측치와 변수 타입**을 확인해 보세요.


In [ ]:
# 1. head() 함수를 사용하여 train, test, submission 데이터의 처음 5행을 확인해 보세요.


In [ ]:
# 2. info() 함수를 사용하여 train, test, submission 데이터의 결측치와 변수 타입을 확인해 보세요.


### Hint.
- <font color="orange">head()</font> : 처음 5행 선택    
    - <font color="orange">(숫자)</font>를 넣으면 지정 숫자만큼의 행을 선택할 수 있습니다.

- <font color="orange">info()</font> : 데이터 결측치 및 변수들의 타입 확인


### Solution.

In [9]:
# 1. head() 함수를 사용하여 train, test, submission 데이터의 처음 5행을 확인해 보세요.
train.head()

index gender car reality  child_num  income_total           income_type  \
0      0      F   N       N          0      202500.0  Commercial associate   
1      1      F   N       Y          1      247500.0  Commercial associate   
2      2      M   Y       Y          0      450000.0               Working   
3      3      F   N       Y          0      202500.0  Commercial associate   
4      4      F   Y       Y          0      157500.0         State servant   

                        edu_type     family_type           house_type  \
0               Higher education         Married  Municipal apartment   
1  Secondary / secondary special  Civil marriage    House / apartment   
2               Higher education         Married    House / apartment   
3  Secondary / secondary special         Married    House / apartment   
4               Higher education         Married    House / apartment   

   DAYS_BIRTH  DAYS_EMPLOYED  FLAG_MOBIL  work_phone  phone  email  \
0      -13899          -4709           1           0      0      0   
1      -11380          -1540           1           0      0      1   
2      -19087          -4434           1           0      1      0   
3      -15088          -2092           1           0      1      0   
4      -15037          -2105           1           0      0      0   

    occyp_type  family_size  begin_month  credit  
0          NaN          2.0         -6.0     1.0  
1     Laborers          3.0         -5.0     1.0  
2     Managers          2.0        -22.0     2.0  
3  Sales staff          2.0        -37.0     0.0  
4     Managers          2.0        -26.0     2.0

In [10]:
test.head()

index gender car reality  child_num  income_total           income_type  \
0  26457      M   Y       N          0      112500.0             Pensioner   
1  26458      F   N       Y          0      135000.0         State servant   
2  26459      F   N       Y          0       69372.0               Working   
3  26460      M   Y       N          0      112500.0  Commercial associate   
4  26461      F   Y       Y          0      225000.0         State servant   

                        edu_type     family_type         house_type  \
0  Secondary / secondary special  Civil marriage  House / apartment   
1               Higher education         Married  House / apartment   
2  Secondary / secondary special         Married  House / apartment   
3  Secondary / secondary special         Married  House / apartment   
4               Higher education         Married  House / apartment   

   DAYS_BIRTH  DAYS_EMPLOYED  FLAG_MOBIL  work_phone  phone  email  \
0      -21990         365243           1           0      1      0   
1      -18964          -8671           1           0      1      0   
2      -15887           -217           1           1      1      0   
3      -19270          -2531           1           1      0      0   
4      -17822          -9385           1           1      0      0   

   occyp_type  family_size  begin_month  
0         NaN          2.0        -60.0  
1  Core staff          2.0        -36.0  
2    Laborers          2.0        -40.0  
3     Drivers          2.0        -41.0  
4    Managers          2.0         -8.0

각 컬럼은 **문자열, 정수, 실수** 등 다양하게 구성되어 있네요!    
test 데이터는 예측해야할 컬럼인 <font color="green">credit</font>가 없습니다.

In [11]:
submission.head()

index  0  1  2
0  26457  0  0  0
1  26458  0  0  0
2  26459  0  0  0
3  26460  0  0  0
4  26461  0  0  0

submission 데이터에서 test 데이터와 동일한 index 컬럼이 보입니다.    
test 데이터로 예측한 <font color="green">credit</font>를 submission에 입력하면 되겠네요.

In [12]:
# 2. info() 함수를 사용하여 train, test, submission 데이터의 결측치와 변수 타입을 확인해 보세요.
train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26457 entries, 0 to 26456
Data columns (total 20 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   index          26457 non-null  int64  
 1   gender         26457 non-null  object 
 2   car            26457 non-null  object 
 3   reality        26457 non-null  object 
 4   child_num      26457 non-null  int64  
 5   income_total   26457 non-null  float64
 6   income_type    26457 non-null  object 
 7   edu_type       26457 non-null  object 
 8   family_type    26457 non-null  object 
 9   house_type     26457 non-null  object 
 10  DAYS_BIRTH     26457 non-null  int64  
 11  DAYS_EMPLOYED  26457 non-null  int64  
 12  FLAG_MOBIL     26457 non-null  int64  
 13  work_phone     26457 non-null  int64  
 14  phone          26457 non-null  int64  
 15  email          26457 non-null  int64  
 16  occyp_type     18286 non-null  object 
 17  family_size    26457 non-null  float64
 18  begin_

**RangeIndex: 26457 entries, 0 to 26456**
▶ train 데이터의 개수는 총 26457개입니다.

**Data columns (total 20 columns):**
▶ 20개의 컬럼으로 구성되어 있고요,

**\#   Column**
▶ 각 컬럼 정보는 데이터 명세와 동일하네요.

**16  occyp_type     18286 non-null  object**
▶ occyp_type 컬럼에 일부 결측치가 존재합니다.

**dtypes: float64(4), int64(8), object(8)**
▶ 데이터 타입은 정수형, object, 실수형으로 다양하게 구성되어 있습니다.

In [14]:
test.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 19 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   index          10000 non-null  int64  
 1   gender         10000 non-null  object 
 2   car            10000 non-null  object 
 3   reality        10000 non-null  object 
 4   child_num      10000 non-null  int64  
 5   income_total   10000 non-null  float64
 6   income_type    10000 non-null  object 
 7   edu_type       10000 non-null  object 
 8   family_type    10000 non-null  object 
 9   house_type     10000 non-null  object 
 10  DAYS_BIRTH     10000 non-null  int64  
 11  DAYS_EMPLOYED  10000 non-null  int64  
 12  FLAG_MOBIL     10000 non-null  int64  
 13  work_phone     10000 non-null  int64  
 14  phone          10000 non-null  int64  
 15  email          10000 non-null  int64  
 16  occyp_type     6848 non-null   object 
 17  family_size    10000 non-null  float64
 18  begin_m

#### QUIZ     
test.info()도 train.info()와 비슷하게 구성되어 있습니다.    
위에서 확인한 해석을 토대로 **해석을 도전**해보세요!

**RangeIndex: 5463 entries, 0 to 5463**
▶ test 데이터의 개수는 총 0,000개입니다.

**Data columns (total 12 columns):**
▶ 00개의 컬럼으로 구성되어 있고요,

**\#   Column**
▶ 각 컬럼 정보는 데이터 명세와 동일하네요.

**1216553 non-null**
▶ 전체 데이터 모두 결측치가 없습니다.

**dtypes: float64(1), int64(5), object(6)**
▶ 데이터 타입은 00형, 000000, 00형으로 다양하게 구성되어 있습니다.

----
#### ANSWER    

**RangeIndex: 5463 entries, 0 to 5463** ▶ test 데이터의 개수는 총 5,463개입니다.

**Data columns (total 12 columns)**: ▶ 12개의 컬럼으로 구성되어 있고요,

**\# Column** ▶ 각 컬럼 정보는 데이터 명세와 동일하네요.

**5463 non-null** ▶ 전체 데이터 모두 결측치가 없습니다.

dtypes: float64(1), int64(5), object(6) ▶ 데이터 타입은 정수형, object, 실수형으로 다양하게 구성되어 있습니다.

In [15]:
submission.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 4 columns):
 #   Column  Non-Null Count  Dtype
---  ------  --------------  -----
 0   index   10000 non-null  int64
 1   0       10000 non-null  int64
 2   1       10000 non-null  int64
 3   2       10000 non-null  int64
dtypes: int64(4)
memory usage: 312.6 KB


# 4. 제출 준비하기

이제 제출을 진행하겠습니다.

초기 상태의 submission.csv 파일을 제출하여 **리더보드**에 올라가 봅시다.  

이후 스테이지를 거듭하며 조금씩 점수를 갱신해보겠습니다.

아래와 같이 to_csv 메소드를 이용하여 submission을 **submission.csv**로 저장해 제출할 수 있습니다.

<font color=red>submission</font><font color=orange>.to_csv</font><font color=black>(</font><font color=green>'beseline.csv',</font><font color=blue>index=False</font><font color=black>)</font>     

각 기능을 색깔로 구별해 볼까요? </br>     

<font color=red>데이터셋이 있는 객체명</font><font color=orange>.객체가 가지고 있는 csv 파일 생성용 매소드(함수)</font><font color=black>(</font><font color=green>저장할 파일의 이름과 확장자,</font><font color=blue>인덱스를 설정하지 않겠음(False)</font><font color=black>)</font>

### Inst.

기본적으로 **".to_csv" 메소드는 데이터프레임(ex. submission) 객체의 내용을 csv 파일 형태로 만들어 저장해 주는 기능**입니다. </br>
이렇게 해야 비로소 파이썬 안에서 작업한 내용이 파이썬 밖으로 빠져나가 여러분이 만질 수 있는 형태가 되는 거죠. </br>
이처럼 파일을 불러서 처리한 후, 저장하는 일련의 과정을 "데이터 i/o(input/output)를 처리한다" 라고 합니다. </br>

그럼 **to_csv 메소드를 이용하여 submission 객체의 내용을 csv 파일 형태**로 만들어 주고,   
index=False를 하여 추가적인 인덱스 설정을 하지 말아 보겠습니다.    

In [ ]:
# to_csv 메소드를 이용하여 submission을 submission.csv로 저장해보세요.


### Hint.
**정확한 채점을 위해서는 index=False를 꼭 넣어 주어야 합니다.**

### Solution.

In [16]:
# to_csv 메소드를 이용하여 submission을 submission.csv로 저장해보세요.
submission.to_csv('submission.csv', index=False)